chiselFrontend/src/main/scala/chisel3/core/Aggregate.scala

```diff
@@ -702,16 +712,17 @@ class Bundle(implicit compileOptions: CompileOptions) extends Record {
       case Some((_, outerInstance)) => mirror.reflect(outerInstance).reflectClass(classSymbol)
       case None => mirror.reflectClass(classSymbol)
     }
-    val clone = classMirror.reflectConstructor(ctor).apply(ctorParamsVals:_*).asInstanceOf[this.type]
-    clone.outerModule = this.outerModule
-
-    if (!clone.typeEquivalent(this)) {
-      reflectError(s"Automatically cloned $clone not type-equivalent to base $this." +
-          " Constructor argument values were inferred: ensure that variable names are consistent and have the same value throughout the constructor chain," +
-          " and that the constructor is deterministic.")
+    val constructor = classMirror.reflectConstructor(ctor)
+    () => {
+      val clone = constructor.apply(ctorParamsVals:_*).asInstanceOf[this.type]
+
+      if (!clone.typeEquivalent(this)) {
+        reflectError(s"Automatically cloned $clone not type-equivalent to base $this." +
```

Comment:
Don't you handle the error messages in the cloneType (instead of the autoCloneType) function? I think this will still do the right thing (this message will take priority because it fires first, as it should), but it's not great from a code structure / separation of concerns standpoint

Reply:
I'll think about how to factor this in the best way